New method: union (111 B)

BabkinAleksandr opened the pull request:
[Ramda union](http://ramdajs.com/docs/#union)

<img width="250" alt="aleksandrbabkin_aleksandrs-macbook-pro____documents_nanoutils" src="https://user-images.githubusercontent.com/25172855/36496256-c8e75f60-1748-11e8-8e33-6d5aa0a5ee0c.png">

Beraliv commented on lib/union/union.test.js:
```diff
@@ -0,0 +1,24 @@
+	const result = ['1', func, 2, 3, false, 1, 4, func2]
+
+	expect(union(a, b)).toEqual(result)
+	expect(union(a)(b)).toEqual(result)
```
add `union(b, a)` and `union(b)(a)` as well

BabkinAleksandr commented:
Added tests and refactored method: in previous version, if we have duplicate values in first array - them would not be filtered, because it was initial value of `reduce` method:

```js
return arr2.reduce(function(acc, item) {
    return acc.indexOf(item) === -1
      ? acc.concat(item)
      : acc
  }, arr1) // here
```

Beraliv commented on lib/union/index.js:
```diff
@@ -1,9 +1,9 @@
       ? acc.concat(item)
       : acc
-  }, arr1)
+  }, [])
```
what's the reason?

BabkinAleksandr replied:
https://github.com/Kelin2025/nanoutils/pull/35#issuecomment-367583126
